src/sage/combinat/integer_vector.py

```diff
@@ -898,6 +900,85 @@ def __contains__(self, x):
             return False
         return sum(x) == self.n
 
+    def rank(self, x):
+        """
+        Return the rank of a given element.
+
+        INPUT:
+
+        - ``x`` -- a list with ``sum(x) == n``
+
+        EXAMPLE::
+
+            sage: IntegerVectors(n=5).rank([5,0])
+            1
+            sage: IntegerVectors(n=5).rank([3,2])
+            3
+            
+        """
+        if sum(x) != self.n:
+            raise ValueError("argument is not a member of IntegerVectors({},{})".format(self.n, None))
+            
+        n, k, r = self.n, len(x), 0
+
+        r = sum(binomial(n + i - 1, i - 1) for i in range(1, k))
```

Comment:
Sorry for not realizing earlier, but this can be simplified further to `r = binomial(k+n-1, n+1)`, if I'm not mistaken.